cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -860,6 +871,50 @@ def set_args(self, request, verbose=False):
 
         self.args = args
 
+    def get_request_files_dir(self):
+        request_files_dir = FilePath(file_dir='request_files')
+        request_files_dir.mkdir()
+        ownership_file_path = os.path.join(request_files_dir.path, '.file_ownerships.json')
+        if not os.path.exists(ownership_file_path):
+            with open(ownership_file_path, 'w') as ownership_file:
+                json.dump({}, ownership_file)
+        return request_files_dir.path
+
+
+    def update_ownership_files(self, uploaded_files_obj):
+        if hasattr(self, 'decoded_token') and self.decoded_token is not None:
+            user_email = tokenHelper.get_token_user_email_address(self.decoded_token)
+            user_roles = tokenHelper.get_token_roles(self.decoded_token)
+        else:
+            user_email = 'public'
+            user_roles = []
+
+        update_file = False
+        ownership_file_path = os.path.join(self.request_files_dir, '.file_ownerships.json')
```

Comment:
Why hide the file?

```suggestion
        ownership_file_path = os.path.join(self.request_files_dir, 'file_ownerships.json')
```

Reply:
This might not be safe if many updates happen for the same request. Can you make separate metdata file for each data file?

Reply:
a metadata file per file is now created